website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+forgot to update our vertex layout definition.
+
+In our `VertexAttribPointer` calls, we declared a size of 3 and a stride of 3 for the `aPosition` attribute.
+Because of this, our buffer is being read like this:
+
+![Buffer reading wrong data](../../../images/opengl/chapter1/leasson3/wrong-pointers.png)
+
+To fix this, we just need to go back to our `VertexAttribPointer()`, which is below our shader compiling section, and
+change it to:
+```c#
+//                                        3 position + 2 texture coordinates! \/
+_gl.VertexAttribPointer(positionLoc, 3, VertexAttribPointerType.Float, false, 5 * sizeof(float), (void*)0);
+```
+
+This will make the buffer be read like that (including the texture coordinate pointer):
+![Buffer reading right data](../../../images/opengl/chapter1/leasson3/right-pointers.png)
+
+
+And if we run the program now, we can see our quad again!
+
+## Using texture coordinates
+Now, we just need to prepare the shader to receive the texture coordinate values. For that, we need to modify both our shaders.
+
+Vertex Shader:
+```
+#version 330 core
+
+layout (location = 0) in vec3 aPosition;
+layout (location = 1) in vec2 aTextureCoord;    // < Add this
+
+//This line stores the data that we want to be received by the fragment
+out vec2 frag_texCoords;                        // < This
+
+void main()
+{
+    gl_Position = vec4(aPosition, 1.0);
+    frag_texCoords = aTextureCoord;             // < And this too
+    // Remember to assigin the data!
+}";
+```
+
+Fragment Shader:
+```
+#version 330 core
+
+// frag_texCoords comes from vertex shader!
+in vec2 frag_texCoords; // < Add this
+
+out vec4 out_color;
+
+void main()
+{
+    // This will allow us to see the texture coordinates in action!
+    out_color = vec4(frag_texCoords.x, frag_texCoords.y, 0, 1.0);
+}
+```
+
+Now we just need to assign the correct layout for `aTextureCoord`. You should add these lines below our first `VertexAttribPointer` call:
+
+```c#
+const uint texCoordLoc = 1;
+_gl.EnableVertexAttribArray(texCoordLoc);
+_gl.VertexAttribPointer(texCoordLoc, 2, VertexAttribPointerType.Float, false, 5 * sizeof(float), (void*)(3 * sizeof(float)));
+```
+
+Pay attention to the last parameter, the pointer! It represents the size of bytes that should be jumped at the start of
+each stride. It means GL will add 3 indexes after stride and pick the consecutive 2 values.
+if you do everything right, you will see this result!
+
+![Quad with UVs](../../../images/opengl/chapter1/leasson3/quad-with-uvs.png)
+
+But we aren't there just yet. We can't see any texture in our quad!
+What you are seeing is the texture coordinates we passed displayed as a color!
+
+What does this mean? The colors that you see are the texture coordinate values of that pixel. As a way to demonstrate it, we're converting
+the numeric values to color values to see what we are doing. In the shader, the values are being read like this:
```

Comment:
```suggestion
The Red and Green colors you see are the X and Y (respectively) texture coordinate values of that pixel. In the shader, the values are being read like this:
```

Reply:
I preferred not to use "red" and "green" because there are not just these colors, the fragment shader interpolates the colors and the reader maybe not understand orange as red+green

Reply:
I added a larger suggestion updating this text and explaining the image